libkbfs/cr_chains.go

```diff
@@ -799,3 +802,66 @@ func (ccs *crChains) changeOriginal(oldOriginal BlockPointer,
 	}
 	return nil
 }
+
+// getPaths returns a sorted slice of most recent paths to all the
+// nodes in the given CR chains that were directly modified during a
+// branch, and which existed at both the start and the end of the
+// branch.  This represents the paths that need to be checked for
+// conflicts.  The paths are sorted by descending path length.  It
+// uses the corresponding node cache when looking up paths, which must
+// at least contain the most recent root node of the branch.  Note
+// that if a path cannot be found, the corresponding chain is
+// completely removed from the set of CR chains.
+func (ccs *crChains) getPaths(ctx context.Context, blocks *folderBlockOps,
```

Comment:
It's a shame that the lightweight `crChains` type now takes a big heavy `folderBlockOps` object as a param. Since `blocks` is only used for `SearchForNodes`, maybe make a one-off interface for that and use that instead? I suggest the name `NodeFinder` or something.